### PR TITLE
Add io.github.wiiznokes.fan-control

### PR DESCRIPTION
**Pull request by wiiznokes**
### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly. [see](https://github.com/wiiznokes/fan-control)
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)* I'm using /sys to write and read sensors values.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: com.wiiznokes.fan-control.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        "install -Dm644 ./resource/linux/com.wiiznokes.fan-control.metainfo.xml /app/share/metainfo/com.wiiznokes.fan-control.metainfo.xml",
+        "install -Dm644 ./resource/linux/com.wiiznokes.fan-control.desktop /app/share/applications/com.wiiznokes.fan-control.desktop",
+        "mkdir -pm755 /app/share/com.wiiznokes.fan-control/resource",
+        "cp -R ./resource/icons /app/share/com.wiiznokes.fan-control/resource"
+      ],
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/wiiznokes/fan-control",
+          "branch": "flatpak"
```
Comment (hfiguiere) [inline com.wiiznokes.fan-control.json:46]: never a git a branch. you need a specific tag / commit

File: com.wiiznokes.fan-control.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{
+  "app-id": "com.wiiznokes.fan-control",
```
Comment (hfiguiere) [inline com.wiiznokes.fan-control.json:2]: the domain `wiiznokes.com` doesn't exist. therefor the app id is invalid, despite you claim the domain is controlled by the upstream developers.

Comment (hfiguiere) [reply com.wiiznokes.fan-control.json:2]: > The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines](https://docs.flathub.org/docs/for-app-authors/requirements#application-id) are followed.

Comment (wiiznokes) [reply com.wiiznokes.fan-control.json:2]: Ho mb, what should i use instead ? I don't have a website for this project, and don't plan to add one for now

Comment (wiiznokes) [reply com.wiiznokes.fan-control.json:2]: does `io.github.wiiznokes.fan-control` would work ?

Comment (wiiznokes) [reply com.wiiznokes.fan-control.json]: changed to `io.github.wiiznokes.fan-control`

**Comment by hfiguiere** [comment]
bot, build io.github.wiiznokes.fan-control

**Comment by hfiguiere** [comment]
```
        "E: io.github.wiiznokes.fan-control:~: releases-info-missing"
```

AppStream file is missing the mandatory `<releases>`

**Comment by hfiguiere** [comment]
Also Rust build fails on aarch64.

**Comment by wiiznokes** [comment]
> Also Rust build fails on aarch64.

Do you know what's the environment for aarch64? I imagine it's a aarch64 machine. Or maybe it's using cross-compilation.

This is maybe related with https://github.com/meta-rust/meta-rust/issues/266, which from my understanding, cause `proc_macro` (`proc_macro_error` in this case), to not be linked, because it is already compiled or something like this.

This will be very hard to fix for me because i don't have an ARM machine, and cross compiling my app seems to work on my machine. 

Anyway, i can try to add some commands to add logs, set some `RUSTFLAGS`, but i don't expect it to fix this.

This would really help if i can only package x86_64, if it's possible

**Comment by hfiguiere** [comment]
> Or maybe it's using cross-compilation.

Flatpak doesn't work for cross-compilation.

**Comment by wiiznokes** [comment]
It should be good now, i disabled ARM. No need to wast CI on this for now

**Comment by bbhtt** [comment]
bot, build io.github.wiiznokes.fan-control

**Pull request closed without merging**